lightkurve/lightcurve.py

```diff
@@ -1087,3 +1089,110 @@ def opt_depth(to, width, data):
         snr_d.append(depth_star * np.sqrt(width_star))
 
     return log_posterior, trial_periods, trial_periods[np.argmax(log_posterior)]
+
+class LightCurveCollection(object):
+    """
+    Collects multiple LightCurve objects together with helpful functions.
+    """
+    def __init__(self, lcs):
+        try:
+            lightcurves = np.asarray(lcs)
+        except:
+            raise TypeError("Unable to parse input")
+        self.data = {}
+        
+        for lc in lightcurves:
+            if isinstance(lc, KeplerLightCurve) or isinstance(lc,LightCurve):
+                if lc.keplerid:
+                    if lc.keplerid in self.data:
+                        self.data[lc.keplerid].append(lc)
+                    else:
+                        self.data[lc.keplerid] = [lc]
+                else: #no keplerID
+                    self.data[None].append(lc)
+            else:
+                raise TypeError("Object is not a LightCurve instance")
+        self.data[None] = []
+
+    def __len__(self):
+        length = 0
+        for lc_array in self.data.values():
+            for lc in lc_array:
+                length += 1
+        return length
+
+    def _ids(self):
+        """
+        Returns the kepler_ids of all the lightcurves as a dict_keys obj.
+        """
+        return self.data.keys()
+
+    def __getitem__(self, kep_id):
+        """
+        Returns the lightcurve associated with the kepler_id. 
+        """
+        try: 
+            return self.data[kep_id]
+        except:
+            raise ValueError('No LightCurve for ' + kep_id)
```

Comment:
This is going to be a pain to extract just one of the `LightCurve` objects. What if I want to do

```
for lightcurve in lightcurvecollection:
    lc.normalize().fold().bin().plot()
```

It will break when I get back multiple `LightCurve`'s based on KeplerID? Will they all be in a list, or be a new collection? I think we need LCC to be iterable in this way.

I think this is what we need:
1. A way to index into the LCC using a simple integer (0...n)
2. If called in a loop, LCC should iterate through every `LightCurve` in the collection.
3. If indexed by KeplerID, it should return another `LightCurveCollection` that has only the `LightCurve`s with the specified ID.

How does that sound?

Reply:
I can achieve this with the current LCC class setup, but it will require some tweaking....
What I'm thinking is to override the __iter__() method so that we can get this kind of functionality for looping you're talking about. I can also accomplish (1, 3) by changing the __getitem__() method to return a new LCC.
Thoughts?